netket/optimizer/qgt/qgt_jacobian_pytree.py

```diff
@@ -112,26 +131,4 @@ def to_dense(self) -> jnp.ndarray:
         Returns:
             A dense matrix representation of this S matrix.
         """
-        if scale is None:
-            O = self.O
-            diag = jnp.eye(self.O.shape[1])
-        else:
-            O = self.O * self.scale[jnp.newaxis, :]
-            diag = jnp.diag(self.scale ** 2)
-
-        return sum_inplace(O.T.conj() @ O) + self.diag_shift * diag
-
-
-def _grad_vmap_minus_mean(
-    fun: Callable, params: jnp.ndarray, samples: jnp.ndarray, holomorphic: bool
-):
-    """Calculates the gradient of a neural network for a number of samples
-    efficiently using vmap(grad),
-    and subtracts their mean for each parameter, i.e., each column
-    """
-    grads = jax.vmap(
-        jax.grad(fun, holomorphic=holomorphic), in_axes=(None, 0), out_axes=0
-    )(params, samples)
-    return grads - sum_inplace(grads.sum(axis=0, keepdims=True)) / (
-        grads.shape[0] * n_nodes
-    )
+        raise NotImplementedError
```

Comment:
More like `CantBeArsedToImplementThisError`

Reply:
take my very lazy boring implementation from onthefly

Reply:
Turns out this needs information about the number of parameters, so that's an argument in favour of adding `params` or extending `scale` as above